### PR TITLE
fix: add necessary comment in generated userdata files in migration

### PR DESCRIPTION
**Pull request by frankqianms**
E2E TEST: N/A
fix bug: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/17956993

### CODE REVIEW

**Review by xzf0587 — DISMISSED** [review]

File: packages/fx-core/src/common/versionMetadata.ts [hunk 1]
```diff
@@ -18,6 +18,8 @@ export const MetadataV3 = {
   teamsManifestFileName: "manifest.json",
   aadManifestFileName: "aad.manifest.json",
   v3UpgradeWikiLink: "https://aka.ms/teams-toolkit-5.0-upgrade",
+  secretFileComment:
```
Comment (blackchoey) [inline packages/fx-core/src/common/versionMetadata.ts:21]: please also add comment for `.env.dev` and `.env.local`, which has different strings.

Comment (frankqianms) [reply packages/fx-core/src/common/versionMetadata.ts:29]: done

File: packages/fx-core/src/common/versionMetadata.ts [hunk 2]
```diff
@@ -18,6 +26,13 @@ export const MetadataV3 = {
...
+    "# This file includes environment variables that will not be committed to git by default. You can set these environment variables in your CI/CD system for your project." +
+    EOL,
+  secretComment:
+    "# Secrets. Keys prefixed with `SECRET_` will be masked in Teams Toolkit logs." + EOL,
+  envFileComment:
```
Comment (blackchoey) [inline packages/fx-core/src/common/versionMetadata.ts:34]: .env.local has different comment than .env.dev, see https://github.com/OfficeDev/TeamsFx/blob/dev/templates/scenarios/js/sso-tab/env/.env.local.tpl#L1

Comment (frankqianms) [reply packages/fx-core/src/common/versionMetadata.ts:34]: got it

**Review by blackchoey — APPROVED** [review]

**Review by xzf0587 — APPROVED** [review]

File: packages/fx-core/src/common/versionMetadata.ts [hunk 3]
```diff
@@ -1,6 +1,14 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
+/**
+ * @author xzf0587 <zhaofengxu@microsoft.com>
+ */
+import { EOL } from "os";
+
+/**
+ * @author xzf0587 <zhaofengxu@microsoft.com>
```
[the remark this answers is not in the document]
Comment (frankqianms) [reply packages/fx-core/src/common/versionMetadata.ts:10]: It is added automatically, I don't know if the format is correct

**Review by xzf0587 — APPROVED** [review]

**Pull request merged**